packages/kbn-test/src/mocha/junit_report_generation.js

```diff
@@ -136,7 +136,13 @@ export function setupJUnitReportGeneration(runner, options = {}) {
       }
     });
 
-    const reportPath = resolve(rootDirectory, `target/junit/TEST-${reportName}.xml`);
+    const reportPath = resolve(
+      rootDirectory,
+      'target/junit',
+      process.env.JOB || '.',
```

Comment:
In what case would the job be undefined? Should we instead use a random string to avoid anything being overwritten?

Reply:
If you were running the tests outside of CI, e.g. locally.

Random string might be helpful when we go back to removing ciGroups / changing how the test suites are run. Can we revisit then, since we'll probably have to revisit anyway?

Reply:
Works for me.